svm/src/transaction_processor.rs

```diff
@@ -602,9 +606,8 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
         //
         // Note these checks are *not* obviated by fee-only transactions.
         let nonce_is_valid = account_loader
-            .load_account(nonce_info.address(), true)
-            .and_then(|loaded_nonce| {
-                let current_nonce_account = &loaded_nonce.account;
+            .load_account(nonce_info.address())
```

Comment:
can you explain why this doesn't use the `load_transaction_account` with inspection? afaict it used to do it

Reply:
ill add another comment here but from the comment on fee-payer:
```
        // We *must* use load_transaction_account() here because *this* is when the fee-payer
        // is loaded for the transaction. Transaction loading skips the first account and
        // loads (and thus inspects) all others normally.
```

we didnt really need to inspect as writable, we just did it for convention because we knew it would be writable. if its a nonce fee-payer its inspected as the fee-payer, if its not its inspected as a normal transaction account in `load_transaction()`

Reply:
[eb6c4f3](https://github.com/anza-xyz/agave/pull/4680/commits/eb6c4f3e913ddba963a4cf8c16552a433c7788bd)